src/messaging/ExchangeContext.cpp

```diff
@@ -84,6 +85,12 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
 
     VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INTERNAL);
 
+    // TODO(#6220): We add this check to make platforms that does not implemented full platform (i.e. Android) can compile, and does
+    // not means that those platforms do not need this.
```

Comment:
I _think_ this is trying to say:
```suggestion
    // TODO(#6220): Some platforms do not implement the full platform and would not compile if 
    // we tried to use a CHIPPlatformLock there.  Those platforms should be fixed so we can remove this
    // #if.
```
but if not, what is it trying to say?

Reply:
@bzbarsky-apple No, un-encryped one does not go through SecureSessionMgr, but also not all messages go through ExchangeContext::SendMessage.   

If we put lock within ExchangeContext::SendMessage, we serialize the send operations of all exchanges. I was wondering if we should do the lock/unlock at lower layer instead of exchange layer.

Reply:
Right, we need to figure out what we're actually trying to serialize here....

Reply:
I think we only need to protect CRMP "send and cache" logic based on the root cause of this crash. (If it is because we try to remove a buffer before it is cached into retransTable)